odoo/models.py

```diff
@@ -1097,37 +1093,8 @@ def _log(base, record, field, exception):
 
     def _validate_fields(self, field_names):
         field_names = set(field_names)
-
-        # old-style constraint methods
-        trans = self.env['ir.translation']
-        errors = []
-        for func, msg, names in self._constraints:
-            try:
-                # validation must be context-independent; call ``func`` without context
-                valid = names and not (set(names) & field_names)
-                valid = valid or func(self)
-                extra_error = None
-            except Exception as e:
-                _logger.debug('Exception while validating constraint', exc_info=True)
-                valid = False
-                extra_error = tools.ustr(e)
-            if not valid:
-                if callable(msg):
-                    res_msg = msg(self)
-                    if isinstance(res_msg, tuple):
-                        template, params = res_msg
-                        res_msg = template % params
-                else:
-                    res_msg = trans._get_source(self._name, 'constraint', self.env.lang, msg)
-                if extra_error:
-                    res_msg += "\n\n%s\n%s" % (_('Error details:'), extra_error)
-                errors.append(res_msg)
-        if errors:
-            raise ValidationError('\n'.join(errors))
-
-        # new-style constraint methods
         for check in self._constraint_methods:
-            if set(check._constrains) & field_names:
+            if not field_names.isdisjoint(check._constrains):
```

Comment:
shouldn't be `if not field_names.isdisjoint(set(check._constrains)):`? 🤔

Reply:
[no](https://docs.python.org/3/library/stdtypes.html?highlight=isdisjoint#frozenset.isdisjoint)

Reply:
But I mean, we should make sure `check._constrains` is a `set` before that `if`, right?

Reply:
Why? `isdisjoint` [accepts any kind of iterable as argument](https://github.com/python/cpython/blob/3e54b575313c64f541e98216ed079fafed01ff5d/Objects/setobject.c#L1393)

Reply:
ah, ok! 😅 thanks